### ISSUE TITLE
getGPS fails with errors

### ISSUE THREAD

**Issue by plorch** [MotusDev/Motus-TO-DO#884]
I have been struggling to get `getGPS()` to work.
When I use the following from Rbook on my data
```
gps_index <- getGPS(sql.motus, 
                    data = df.alltags.sub, 
                    by = "closest")
```
I get 

Error: Problem with `mutate()` input `gpsID`.
x can only subtract from "POSIXt" objects
i Input `gpsID` is `purrr::map2_int(...)`.

Removing all NA from tagDeployStart and tagDeployEnd due to bad deployments does not help.  Neither does defining a cutoff.

Similarly the same as above with ` by = "closest"` gives

Error: Problem with `mutate()` input `timeBin`.
x '/' not defined for "POSIXt" objects
i Input `timeBin` is `as.integer(.data$ts/by)`.

The problem seems to be that you need to convert the ts coming from the src tbl needs to be made into a POSIXt with `as_datetime` in `motus:::prepGPS()`

**Comment by denislepage** [comment]
@steffilazerte  could you look into this? 

@plorch could you make sure you have the latest build of the R package?

**Comment by steffilazerte** [comment]
Hi @plorch, 

Right now I can't reproduce this error with Project 4, so I think it may either be specific to your system or to your data. So to figure it out I'll need some more details. 

1. Can you confirm that you're using the most recent version of motus (if not please update):
```
packageVersion("motus")
[1] ‘4.0.3’
```

2. If that doesn't help, what project are you working with? And what filtering have you done to df.alltags.sub? This info will let me test out the problem on my system and see if I can't reproduce the error.

Thanks!

**Comment by plorch** [comment]
1. I can confirm that I am using 4.0.3
2. I am working with PARC data (proj.num = 35) and working through the example 5.5 in the Rbook.

There are three likely sources of problems.    One is that that there are detections without deployment data.  Sadly these are mostly ours, but I cannot fix them.  I am working on that issue, and I think I am filtering them out (see below).  The second thing is that this project spans the time when motusFilter was added, so I need to use `filterByActivity()`.  There is a bug in that function where it returns NA instead of 1 for cases that should be kept (see Issue #871 ).  So to get records that meet the criteria to keep, I have substituted `is.na(probability)` for `probability == 1`  This may be the source of the problem, but I don't see another way to get the good records.

Here is an attempted repex (big project, so first step will take a while):
```
library(motus)
proj.num = 35
proj.sql.motus <- tagme(projRecv = proj.num, new = TRUE,update = TRUE)

tbl.alltags <- filterByActivity(proj.sql.motus, return = "all")

df.alltags.sub <- tbl.alltags %>% 
  filter(is.na(probability)) %>%
  collect() %>%
  as.data.frame() %>%
  mutate(ts = as_datetime(ts),  # work with dates AFTER transforming to flat file
         tagDeployStart = as_datetime(tagDeployStart),
         tagDeployEnd = as_datetime(tagDeployEnd))
df.alltags.sub.1 = df.alltags.sub %>%
  filter(!is.na(tagDeployStart))
tbl.gps = tbl(proj.sql.motus,"gps")
df.gps = tbl.gps %>%
  collect() %>%
  as.data.frame()
```

**Comment by steffilazerte** [comment]
Okay, the quick and dirty fix is to not transform your df.alltags.sub `ts` to datetime (i.e. remove all the `as_datetime()` statements and leave them as numeric timestamps). The problem is that `getGPS()` assumes that the `ts` column is a numeric value, not a datetime value (my bad!). 

I'm testing a proper fix that will permit both numeric and datetime `ts`, but it'll take a day or so. Will report back when I've got it (also looking into #871 and will have a fix for that at the same time)!

You can test this out with a super small subset on only the first 100 observations:
```
getGPS(proj.sql.motus, data = df.alltags.sub[1:100, ], by = "closest")
```

If that works then give the whole data set a go (it will take a while!)

**Comment by steffilazerte** [comment]
Okay this should be fixed in beta4. (Also fixed #871)  To test it out:

```
remotes::install_github("MotusWTS/motus", ref = "beta4")
library(motus)
tags <- tagme(...)
...
```

**Comment by plorch** [comment]
Thanks Steffi.  It works.  I am back on track.  I was able to make a leaflet map to compare paths of birds with different histories.
